### PR TITLE
Stream refactor

### PR DESCRIPTION
**Pull request by ldxdl**
### Issues associated with this PR

Simplify stream layer & enhance performance

### Sign the CLA
Signed

### Solutions
Simplify client interface & impl

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases.
And you need to show the ut result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by neverhook** [comment]
LGTM

File: pkg/stream/http/stream.go [hunk 1]
```diff
@@ -238,20 +240,47 @@ func (csc *clientStreamConnection) NewStream(ctx context.Context, receiver types
...
+}
+
+func (conn *clientStreamConnection) Reset(reason types.StreamResetReason) {
+	conn.mutex.Lock()
+	defer conn.mutex.Unlock()
+
+	if conn.stream != nil {
+		conn.stream.ResetStream(reason)
+
+		conn.mutex.Lock()
```
Comment (neverhook) [inline pkg/stream/http/stream.go]: conn.mutex.Lock()执行了两次，会造成死锁，下面的if中的锁操作可以去掉

Comment (ldxdl) [reply pkg/stream/http/stream.go]: Fixed
